git-cliff-core/src/repo.rs

```diff
@@ -75,37 +85,152 @@ impl Repository {
 			.filter_map(|id| self.inner.find_commit(id).ok())
 			.collect();
 		if include_path.is_some() || exclude_path.is_some() {
+			// Normalize the glob patterns
+			let include_patterns = include_path.map(|patterns| {
+				patterns.into_iter().map(Self::normalize_glob).collect()
+			});
+			let exclude_patterns = exclude_path.map(|patterns| {
+				patterns.into_iter().map(Self::normalize_glob).collect()
+			});
+
 			commits.retain(|commit| {
-				if let Ok(prev_commit) = commit.parent(0) {
-					if let Ok(diff) = self.inner.diff_tree_to_tree(
-						commit.tree().ok().as_ref(),
-						prev_commit.tree().ok().as_ref(),
-						None,
-					) {
-						return diff
-							.deltas()
-							.filter_map(|delta| delta.new_file().path())
-							.any(|new_file_path| {
-								if let Some(include_path) = &include_path {
-									include_path
-										.iter()
-										.any(|glob| glob.matches_path(new_file_path))
-								} else if let Some(exclude_path) = &exclude_path {
-									!exclude_path
-										.iter()
-										.any(|glob| glob.matches_path(new_file_path))
-								} else {
-									false
-								}
-							});
-					}
-				}
-				false
+				self.should_retain_commit(
+					commit,
+					&include_patterns,
+					&exclude_patterns,
+				)
 			});
 		}
 		Ok(commits)
 	}
 
+	/// Normalizes the glob pattern by removing the leading `./`.
+	fn normalize_glob(pattern: Pattern) -> Pattern {
```

Comment:
I feel like this can be moved into `commits` function (as a inner function maybe) since it is not used anywhere + not being tested.

Reply:
It's been used in the tests now as well, so it's better to keep it separate.